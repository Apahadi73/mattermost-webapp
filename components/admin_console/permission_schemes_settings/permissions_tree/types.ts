// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {ClientConfig, ClientLicense} from 'mattermost-redux/types/config.js';

export type Permission = {
    id: string;
    combined?: boolean;
    permissions: string[];
}
export type Group = {
    id: string;
    permissions: Array<Permission | string>;
    isVisible?: (license: (Partial<ClientLicense> | undefined), config: Partial<ClientConfig>, scope: string) => boolean;
}

export type AdditionalValues = {
    edit_post: {
        editTimeLimitButton: JSX.Element;
    };
}
