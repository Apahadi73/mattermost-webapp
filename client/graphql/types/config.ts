// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export interface Config {
    AboutLink: string;
    CloudUserLimit: string;
    DataRetentionBoardsRetentionDays: '0';
    DataRetentionEnableBoardsDeletion: TrueOrFalseInString;
    ExtendSessionLengthWithActivity: TrueOrFalseInString;
    FeatureFlagAddMembersToChannel: 'top';
    FeatureFlagAppBarEnabled: TrueOrFalseInString;
    FeatureFlagAppsEnabled: TrueOrFalseInString;
    FeatureFlagBoardsDataRetention: TrueOrFalseInString;
    FeatureFlagBoardsFeatureFlags: '';
    FeatureFlagCallsMobile: TrueOrFalseInString;
    FeatureFlagCloudDelinquentEmailJobsEnabled: TrueOrFalseInString;
    FeatureFlagCollapsedThreads: TrueOrFalseInString;
    FeatureFlagCustomGroups: TrueOrFalseInString;
    FeatureFlagEnableInactivityCheckJob: TrueOrFalseInString;
    FeatureFlagEnableRemoteClusterService: TrueOrFalseInString;
    FeatureFlagGraphQL: TrueOrFalseInString;
    FeatureFlagGuidedChannelCreation: TrueOrFalseInString;
    FeatureFlagInviteToTeam: 'none';
    FeatureFlagNewAccountNoisy: TrueOrFalseInString;
    FeatureFlagNormalizeLdapDNs: TrueOrFalseInString;
    FeatureFlagPermalinkPreviews: TrueOrFalseInString;
    FeatureFlagPluginApps: '';
    FeatureFlagPluginFocalboard: '';
    FeatureFlagPluginPlaybooks: '';
    FeatureFlagTestBoolFeature: TrueOrFalseInString;
    FeatureFlagTestFeature: 'off';
    FeatureFlagUseCaseOnboarding: TrueOrFalseInString;
    GfycatApiKey: '2_KtH_W5';
    GfycatApiSecret: '3wLVZPiswc3DnaiaFoLkDvB4X0IV6CpMkj4tf2inJRsBY6-FnkT08zGmppWFgeof';
    TelemetryId: '7j4czyms1bgtuq4buyi9eutf9y';
    AllowBannerDismissal: string;
    AllowCustomThemes: string;
    AllowedThemes: string;
    AndroidAppDownloadLink: string;
    AndroidLatestVersion: string;
    AndroidMinVersion: string;
    AppDownloadLink: string;
    AsymmetricSigningPublicKey: string;
    AvailableLocales: string;
    BannerColor: string;
    BannerText: string;
    BannerTextColor: string;
    BuildDate: string;
    BuildEnterpriseReady: string;
    BuildHash: string;
    BuildHashEnterprise: string;
    BuildNumber: string;
    CollapsedThreads: CollapsedThreads;
    CustomBrandText: string;
    CustomDescriptionText: string;
    CustomTermsOfServiceId: string;
    CustomTermsOfServiceReAcceptancePeriod: string;
    CustomUrlSchemes: string;
    CWSURL: string;
    DataRetentionEnableFileDeletion: string;
    DataRetentionEnableMessageDeletion: string;
    DataRetentionFileRetentionDays: string;
    DataRetentionMessageRetentionDays: string;
    DefaultClientLocale: string;
    DefaultTheme: ThemeKey;
    DiagnosticId: string;
    DiagnosticsEnabled: string;
    EmailLoginButtonBorderColor: string;
    EmailLoginButtonColor: string;
    EmailLoginButtonTextColor: string;
    EmailNotificationContentsType: string;
    EnableAskCommunityLink: string;
    EnableBanner: string;
    EnableBotAccountCreation: string;
    EnableChannelViewedMessages: string;
    EnableClientPerformanceDebugging: string;
    EnableCluster: string;
    EnableCommands: string;
    EnableCompliance: string;
    EnableConfirmNotificationsToChannel: string;
    EnableCustomBrand: string;
    EnableCustomEmoji: string;
    EnableCustomGroups: string;
    EnableCustomUserStatuses: string;
    EnableTimedDND: string;
    EnableCustomTermsOfService: string;
    EnableDeveloper: string;
    EnableDiagnostics: string;
    EnableEmailBatching: string;
    EnableEmailInvitations: string;
    EnableEmojiPicker: string;
    EnableFileAttachments: string;
    EnableFile: string;
    EnableGifPicker: string;
    EnableGuestAccounts: string;
    EnableIncomingWebhooks: string;
    EnableLatex: string;
    EnableInlineLatex: string;
    EnableLdap: string;
    EnableLinkPreviews: string;
    EnableMarketplace: string;
    EnableMetrics: string;
    EnableMobileFileDownload: string;
    EnableMobileFileUpload: string;
    EnableMultifactorAuthentication: string;
    EnableOAuthServiceProvider: string;
    EnableOpenServer: string;
    EnableOutgoingWebhooks: string;
    EnablePostIconOverride: string;
    EnablePostUsernameOverride: string;
    EnablePreviewFeatures: string;
    EnablePreviewModeBanner: string;
    EnablePublicLink: string;
    EnableReliableWebSockets: string;
    EnableSaml: string;
    EnableSignInWithEmail: string;
    EnableSignInWithUsername: string;
    EnableSignUpWithEmail: string;
    EnableSignUpWithGitLab: string;
    EnableSignUpWithGoogle: string;
    EnableSignUpWithOffice365: string;
    EnableSignUpWithOpenId: string;
    EnableSVGs: string;
    EnableTesting: string;
    EnableThemeSelection: string;
    EnableTutorial: string;
    EnableOnboardingFlow: string;
    EnableUserAccessTokens: string;
    EnableUserCreation: string;
    EnableUserDeactivation: string;
    EnableUserTypingMessages: string;
    EnforceMultifactorAuthentication: string;
    ExperimentalClientSideCertCheck: string;
    ExperimentalClientSideCertEnable: string;
    ExperimentalCloudBilling: string;
    ExperimentalCloudUserLimit: string;
    ExperimentalDataPrefetch: string;
    ExperimentalEnableAuthenticationTransfer: string;
    ExperimentalEnableAutomaticReplies: string;
    ExperimentalEnableClickToReply: string;
    ExperimentalEnableDefaultChannelLeaveJoinMessages: string;
    ExperimentalEnablePostMetadata: string;
    ExperimentalGroupUnreadChannels: string;
    ExperimentalPrimaryTeam: string;
    ExperimentalTimezone: string;
    ExperimentalViewArchivedChannels: string;
    FileLevel: string;
    GfycatAPIKey: string;
    GfycatAPISecret: string;
    GoogleDeveloperKey: string;
    GuestAccountsEnforceMultifactorAuthentication: string;
    HasImageProxy: string;
    HelpLink: string;
    IosAppDownloadLink: string;
    IosLatestVersion: string;
    IosMinVersion: string;
    InstallationDate: string;
    IsDefaultMarketplace: string;
    LdapFirstNameAttributeSet: string;
    LdapLastNameAttributeSet: string;
    LdapLoginButtonBorderColor: string;
    LdapLoginButtonColor: string;
    LdapLoginButtonTextColor: string;
    LdapLoginFieldName: string;
    LdapNicknameAttributeSet: string;
    LdapPositionAttributeSet: string;
    LdapPictureAttributeSet: string;
    LockTeammateNameDisplay: string;
    ManagedResourcePaths: string;
    MaxFileSize: string;
    MaxPostSize: string;
    MaxNotificationsPerChannel: string;
    MinimumHashtagLength: string;
    NoAccounts: string;
    GitLabButtonText: string;
    GitLabButtonColor: string;
    OpenIdButtonText: string;
    OpenIdButtonColor: string;
    PasswordMinimumLength: string;
    PasswordRequireLowercase: string;
    PasswordRequireNumber: string;
    PasswordRequireSymbol: string;
    PasswordRequireUppercase: string;
    PluginsEnabled: string;
    PostEditTimeLimit: string;
    PrivacyPolicyLink: string;
    ReportAProblemLink: string;
    RequireEmailVerification: string;
    RestrictDirectMessage: string;
    RunJobs: string;
    SamlFirstNameAttributeSet: string;
    SamlLastNameAttributeSet: string;
    SamlLoginButtonBorderColor: string;
    SamlLoginButtonColor: string;
    SamlLoginButtonText: string;
    SamlLoginButtonTextColor: string;
    SamlNicknameAttributeSet: string;
    SamlPositionAttributeSet: string;
    SendEmailNotifications: string;
    SendPushNotifications: string;
    ShowEmailAddress: string;
    SiteName: string;
    SiteURL: string;
    SQLDriverName: string;
    SupportEmail: string;
    TeammateNameDisplay: string;
    TermsOfServiceLink: string;
    TimeBetweenUserTypingUpdatesMilliseconds: string;
    UpgradedFromTE: string;
    Version: string;
    WebsocketPort: string;
    WebsocketSecurePort: string;
    WebsocketURL: string;
    ExperimentalSharedChannels: string;
}

type TrueOrFalseInString = 'true' | 'false';

export enum CollapsedThreads {
    DISABLED = 'disabled',
    DEFAULT_ON = 'default_on',
    DEFAULT_OFF = 'default_off',
}

export type ThemeKey = 'denim' | 'sapphire' | 'quartz' | 'indigo' | 'onyx';
