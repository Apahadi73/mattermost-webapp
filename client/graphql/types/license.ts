// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {TrueOrFalseInString} from './config';

export interface License extends Record<string, string> {
    IsLicensed: TrueOrFalseInString;
    Cloud: TrueOrFalseInString;
}
