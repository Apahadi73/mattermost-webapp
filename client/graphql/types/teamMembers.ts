// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export interface TeamMembers {
    team: Team[];
}

interface Team {
    id: string;
    displayName: string;
    name: string;
    updateAt: number;
    description: string;
    email: string;
    type: string;
    companyName: string;
    allowedDomains: string;
    inviteId: string;
    lastTeamIconUpdate: number;
    groupConstrained: boolean | null;
    allowOpenInvite: boolean;
}
