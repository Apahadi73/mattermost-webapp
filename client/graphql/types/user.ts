// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export interface User {
    id: string;
    username: string;
    email: string;
    firstName: string;
    lastName: string;
    nickname: string;
    isBot: boolean;
    isGuest: boolean;
    isSystemAdmin: boolean;
    createAt: number;
    deleteAt: number;
    authService: string;
    customStatus: CustomStatus;
    status: Status;
    props: Record<string, string>;
    notifyProps: UserNotifyProps;
    lastPictureUpdateAt: number;
    locale: string;
    timezone: UserTimezone;
    position: string;
    roles: Role[];
    preferences: Preference[];
    sessions: Session[];
}

interface CustomStatus {
    emoji: string;
    text: string;
    duration: string;
    expiresAt: string;
}

interface Status {
    status: string;
    manual: number;
    lastActivityAt: number;
    activeChannel: string;
    dndEndTime: number;
}

interface Role {
    id: string;
    name: string;
    permissions: string[];
    schemeManaged: boolean;
    builtIn: boolean;
}

export interface Preference {
    userId: string;
    category: string;
    name: string;
    value: string;
}

interface Session {
    d: string;
    token: string;
    createAt: number;
    expiresAt: number;
}

export type UserNotifyProps = {
    desktop: 'default' | 'all' | 'mention' | 'none';
    desktop_sound: 'true' | 'false';
    email: 'true' | 'false';
    mark_unread: 'all' | 'mention';
    push: 'default' | 'all' | 'mention' | 'none';
    push_status: 'ooo' | 'offline' | 'away' | 'dnd' | 'online';
    comments: 'never' | 'root' | 'any';
    first_name: 'true' | 'false';
    channel: 'true' | 'false';
    mention_keys: string;
};

export type UserTimezone = {
    useAutomaticTimezone: boolean | string;
    automaticTimezone: string;
    manualTimezone: string;
};
