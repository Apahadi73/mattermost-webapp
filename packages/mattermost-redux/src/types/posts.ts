// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Post} from '@mattermost/types/posts';
import {PostTypes} from 'utils/constants';

export type UserActivityPost = Post & {
    system_post_ids: string[];
    user_activity_posts: Post[];
}

export type PostWithFormatData = Post & {
    isFirstReply: boolean;
    isLastReply: boolean;
    previousPostIsComment: boolean;
    commentedOnPost?: Post;
    consecutivePostByUser: boolean;
    replyCount: number;
    isCommentMention: boolean;
    highlight: boolean;
};

// New post is a post that is pending being
// saved on the server
export interface NewPost {
    id: Post['id'];
    user_id: Post['user_id'];
    channel_id: Post['channel_id'];
    message: Post['message'];
    type: typeof PostTypes[keyof typeof PostTypes];
    create_at: number;
    update_at: number;
    root_id: string;
    props: {
        username?: string;
        addedUsername?: string;
        addedUserId?: string;
    };
}
