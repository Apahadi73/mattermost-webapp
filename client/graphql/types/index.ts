// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {User} from '../types/user';
import {Config} from '../types/config';
import {TeamMember} from '../types/teamMembers';

export interface ApiGraphQLTypes {
    user: User;
    config: Config;
    license: Record<string, string>;
    teamMembers: TeamMember[];
}
