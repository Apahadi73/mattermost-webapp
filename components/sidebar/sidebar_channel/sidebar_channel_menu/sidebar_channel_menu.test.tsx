// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';

import {ChannelType} from '@mattermost/types/channels';

import Constants from 'utils/constants';
import {TestHelper} from 'utils/test_helper';

import SidebarChannelMenu from './sidebar_channel_menu';

describe('components/sidebar/sidebar_channel/sidebar_channel_menu', () => {
    const testChannel = TestHelper.getChannelMock();
    const testCategory = TestHelper.getCategoryMock();

    const baseProps = {
        channel: testChannel,
        channelLink: 'http://a.fake.link',
        categories: [testCategory],
        currentUserId: 'user_id',
        currentCategory: testCategory,
        currentTeamId: 'team_id',
        location: 'sidebar',
        isUnread: false,
        isFavorite: false,
        isMuted: false,
        managePublicChannelMembers: true,
        managePrivateChannelMembers: true,
        closeHandler: jest.fn(),
        isCollapsed: false,
        isMenuOpen: true,
        onToggleMenu: jest.fn(),
        multiSelectedChannelIds: [],
        displayedChannels: [],
        actions: {
            markChannelAsRead: jest.fn(),
            favoriteChannel: jest.fn(),
            unfavoriteChannel: jest.fn(),
            muteChannel: jest.fn(),
            unmuteChannel: jest.fn(),
            openModal: jest.fn(),
            createCategory: jest.fn(),
            addChannelsInSidebar: jest.fn(),
        },
    };

    // there is a separate ticket to adjust tests

    test('should match snapshot and contain correct buttons', () => {
        const wrapper = shallow(
            <SidebarChannelMenu {...baseProps}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});