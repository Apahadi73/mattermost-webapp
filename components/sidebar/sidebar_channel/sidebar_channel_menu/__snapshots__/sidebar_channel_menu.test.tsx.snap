// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sidebar/sidebar_channel/sidebar_channel_menu should match snapshot and contain correct buttons 1`] = `
<Fragment>
  <button
    className="SidebarMenu_menuButton"
    onClick={[Function]}
  />
  <Menu
    data={
      Object {
        "items": Array [
          false,
          Object {
            "label": "Favorite",
            "leadingElement": <i
              className="icon-star-outline"
            />,
            "onClick": [Function],
          },
          Object {
            "divider": true,
            "label": "Mute Channel",
            "leadingElement": <i
              className="icon-bell-outline"
            />,
            "onClick": [Function],
          },
          Object {
            "label": "Copy Link",
            "leadingElement": <i
              className="icon-link-variant"
            />,
            "onClick": [Function],
          },
          Object {
            "items": Array [
              Object {
                "label": "category_name",
                "leadingElement": <i
                  className="icon-folder-outline"
                />,
                "onClick": [Function],
              },
              Object {
                "label": "New Category",
                "leadingElement": <i
                  className="icon-plus"
                />,
                "onClick": [Function],
              },
            ],
            "label": "Move to...",
            "leadingElement": <i
              className="icon-folder-move-outline"
            />,
            "onClick": [Function],
            "trailingElement": <i
              className="icon-chevron-right"
            />,
            "trailingElementLabel": "selected",
          },
          Object {
            "label": "Add Members",
            "leadingElement": <i
              className="icon-account-outline"
            />,
            "onClick": [Function],
          },
          Object {
            "label": "Leave Channel",
            "leadingElement": <i
              className="icon-close"
            />,
            "onClick": [Function],
          },
        ],
        "title": "Channels",
      }
    }
    open={false}
    overlayCloseHandler={[Function]}
    title="Channel Menu"
    trigger={
      Object {
        "current": null,
      }
    }
  />
</Fragment>
`;
